golem-test-framework/src/components/component_service/mod.rs

```diff
@@ -85,20 +86,17 @@ pub enum PluginServiceClient {
 }
 
 #[async_trait]
-pub trait ComponentService {
-    fn client_protocol(&self) -> GolemClientProtocol;
+trait ComponentServiceInternal {
```

Comment:
These are only used to implement the defaults in the main trait, so there is no reason to expose them to other parts of the code